src/rgw/rgw_rest_s3.cc

```diff
@@ -6072,10 +6101,16 @@ int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist &bl, off_t ofs, off_
   if (m_s3select_query.find(LT) != std::string::npos)
     boost::replace_all(m_s3select_query, LT, "<");
 
-  size_t _qs = m_s3select_query.find("<Expression>", 0) + strlen("<Expression>");
-  size_t _qe = m_s3select_query.find("</Expression>", _qs);
+  std::string query;
+  
+  //AWS cli s3select parameters
+  extract_by_tag("Expression",query);
+  extract_by_tag("FieldDelimiter",m_column_delimiter);convert_escape_seq(m_column_delimiter);
```

Comment:
odd spacing

Reply:
this "statement;statement" thing is odd too

Reply:
ok